Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -43,7 +43,9 @@ struct WhitespaceContent {
     size_t length { 0 };
     bool isWordSeparator { true };
 };
-static std::optional<WhitespaceContent> moveToNextNonWhitespacePosition(StringView textContent, size_t startPosition, bool preserveNewline, bool preserveTab, bool stopAtWordSeparatorBoundary)
+
+template<typename CharacterType>
+static std::optional<WhitespaceContent> moveToNextNonWhitespacePosition(CharacterType* textContent, unsigned length, size_t startPosition, bool preserveNewline, bool preserveTab, bool stopAtWordSeparatorBoundary)
```

Comment:
In new code I like to use `std::span<const CharacterType>`.

Reply:
I purposefully didn't use std::span<> because its operator[] does a bounds check and this is hot code. 

Note that the reason I am moving this code away from StringView::operator[] is because adding such a bounds check to StringView::operator[] is a performance regression on both Jetstream and Speedometer.

Odds are that by using std::span here I'd introduce the same perf regression I saw when adding the bounds check to  StringView::operator[].

Reply:
Yes, I could have passed a span and used span.data()[index] inside. I'll follow-up after I'm done with some urgent stuff.

Reply:
Following up in https://github.com/WebKit/WebKit/pull/19238.